ext/msgpack/buffer_class.c

```diff
@@ -76,10 +65,21 @@ const rb_data_type_t buffer_data_type = {
     .flags = RUBY_TYPED_FREE_IMMEDIATELY
 };
 
+const rb_data_type_t buffer_view_data_type = {
+    .wrap_struct_name = "msgpack:buffer_view",
+    .function = {
+        .dmark = msgpack_buffer_mark,
+        .dfree = NULL,
+        .dsize = NULL,
+    },
+    .flags = RUBY_TYPED_FREE_IMMEDIATELY
+};
+
 static inline msgpack_buffer_t *MessagePack_Buffer_get(VALUE object)
 {
     msgpack_buffer_t *buffer;
-    TypedData_Get_Struct(object, msgpack_buffer_t, &buffer_data_type, buffer);
+    bool view = RTEST(rb_ivar_get(object, rb_intern("@owner")));
```

Comment:
We can move the `rb_intern` into an a global `ID` variable so we don't need to do lookup every time.

Reply:
Done.